packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -182,10 +182,245 @@ export interface LocalSecondaryIndexProps extends SecondaryIndexProps {
   readonly sortKey: Attribute;
 }
 
+/**
+ * Imported or created dynamodb table
+ */
+export interface ITable extends IResource {
+  /**
+   * arn of the dynamodb table
+   *
+   * @attribute
+   */
+  readonly tableArn: string;
+
+  /**
+   * table name of the dynamodb table
+   *
+   * @attribute
+   */
+  readonly tableName: string;
+
+  /**
+   * stream arn of the dynamodb table
+   *
+   * @attribute
+   */
+  readonly tableStreamArn?: string;
+
+  /**
+   * Adds an IAM policy statement associated with this table to an IAM
+   * principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "dynamodb:PutItem", "dynamodb:GetItem", ...)
+   */
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
```

Comment:
Let's not have `grant` in `ITable` for now (it's not present in `IBucket`, for example). It still should be available in `Table` though, of course.

Reply:
Looks like you are suggesting creating an empty `ITable` abstract without any method. Firstly I find below read-only methods from s3's `IBucket`.

```
urlForObject(key?: string): string;
grantRead(identity: iam.IGrantable, objectsKeyPattern?: any): iam.Grant;
...
```

Secondly, the client uses `Table` instead of `ITable` from imported dynamodb table, the imported table will have the capability of modifying the table schemas or indexes, which depends on the whole construct info of dynamodb.

I think the use case of imported dynamodb is that using high level APIs to grant the dynamodb permission to newly created lambda, role of ec2/service, and so on.

Let me know what you think.

Reply:
> Looks like you are suggesting creating an empty `ITable` abstract without any method. 

No, that is *not* what I'm saying. I'm saying remove the `grant` method. Keep the remaining methods (other than `grantStream`, which I commented on below).